application/comit_node/src/logging.rs

```diff
@@ -29,27 +17,22 @@ pub fn set_up_logging(settings: &ComitNodeSettings) {
 }
 
 fn formatter(out: FormatCallback<'_>, message: &Arguments<'_>, record: &Record<'_>) {
-    // configure colors for the whole line
-    let colors_line = ColoredLevelConfig::default()
-        .info(Color::Green)
-        .debug(Color::Blue)
-        .trace(Color::Cyan);
+    let line = match record.line() {
```

Comment:
:wrench: This can be expressed more idiomatically as:

```rust
let line_number = record.line().map(|line| format!(":{}", line)).unwrap_or_else(|| String::new())
```

Reply:
❓ what makes `map` more idiomatic than `match` ?

Reply:
It probably boils down to personal preference actually :sweat_smile: 
The way I see it is that `match` is like a 'terminal' operation me, like `for_each` on an iterator. Something where I can cause side-effects for example.

But if all I want is to *change* the option and re-assign it, I always try to use the APIs on `Option`, `Result` etc.

It is a bit like doing `for x in list` vs `iter.map`.

The first one allows you to do whatever you want (same as the `match` statement) because you are simply unpacking it whereas the latter describes through the method calls, what is happening.